addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusHandler.java

```diff
@@ -0,0 +1,369 @@
+    private PowerFlowRealtimeResponse powerFlowResponse;
+
+    private Gson gson;
+
+    public FroniusHandler(Thing thing) {
+        super(thing);
+        gson = new Gson();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId());
+        } else {
+            logger.debug("The Air Quality binding is read-only and can not handle command {}", command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Fronius handler.");
+
+        FroniusConfiguration config = getConfigAs(FroniusConfiguration.class);
+
+        boolean validConfig = true;
+        String errorMsg = null;
+        if (StringUtils.trimToNull(config.ip) == null) {
+            errorMsg = "Parameter 'Ip' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (config.refresh != null && config.refresh <= 0) {
+            errorMsg = "Parameter 'refresh' must be at least 1 second";
+            validConfig = false;
+        }
+
+        if (validConfig) {
+            startAutomaticRefresh();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+        }
+    }
+
+    /**
+     * Start the job refreshing the data
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Request
+                        updateData();
+                        // Update all channels
+                        for (Channel channel : getThing().getChannels()) {
+                            updateChannel(channel.getUID().getId());
+                        }
+                    } catch (Exception e) {
+                        logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
+                }
+            };
+
+            FroniusConfiguration config = getConfigAs(FroniusConfiguration.class);
+            int delay = (config.refresh != null) ? config.refresh.intValue() : DEFAULT_REFRESH_PERIOD;
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Fronius handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    /**
+     * Update the channel from the last data
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    private void updateChannel(String channelId) {
+        if (isLinked(channelId)) {
```

Comment:
If you turn this around with an early return you can save one level of indentation fort he rest of this method.
`if(! isLinked(channelId) {return;} ...`